app/pages/notifications/notification-section.jsx

```diff
@@ -117,6 +118,15 @@ export default class NotificationSection extends Component {
     });
   }
 
+  markAllRead() {
+    talkClient.put('/notifications/read', { section: this.props.section });
+    this.state.notificationData.forEach((data) => {
+      data.notification.update({ delivered: true }).save();
+    });
+    this.setState({ unread: 0 });
+    this.context.notificationsCounter.setUnread(0);
```

Comment:
An alternative is to `talkClient.type('notifications').get({ page_size: 1, delivered: false })`, get the count from the meta, and update if zero.